### ISSUE TITLE
[CRITICAL] [Backwards Compatibility] Moving reports between policies

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#33176]
### Problem 

As part of the workspace chats for [paid group policies](https://github.com/Expensify/Expensify/issues/309695), we need to make sure actions possible to take in OldDot are corrently reflected in NewDot.

One of them is moving reports between policies, this influences how the workspace chat report preview is shown.

The high level details are explained in the Design doc section [here](https://docs.google.com/document/d/1arowjiiYSu4fyu0CYrB8748UOz4oEDaWZzmXpGKwg-k/edit#bookmark=id.acotdlrhd15c) in terms of what we need to do and check for when we move the report to a policy with workspace chat or from a policy with workspace chat to some which does not have workspace chat.

Please bring ay questions to the wave6 room, the design doc is not written in a lot of detail but the things we need to do are captured there is plain English

**Comment by madmax330** [comment]
Should be ready early EOW next week

**Comment by madmax330** [comment]
Still working on it

**Comment by greg-schroeder** [comment]
Still held

**Comment by greg-schroeder** [comment]
Removing hold per room discussion!

**Comment by madmax330** [comment]
The part we're holding on will be added in: https://github.com/Expensify/Auth/pull/9626
Then we'll be able to merge the PR for this issue:  https://github.com/Expensify/Auth/pull/9758

**Comment by madmax330** [comment]
PR is on hold for the other pr that adds all the transaction threads and report actions

**Comment by madmax330** [comment]
We had to roll this back due to performance issues

**Comment by madmax330** [comment]
Putting this one on hold since it also depends on unreported transactions so the PR for it is waiting on https://github.com/Expensify/Expensify/issues/357777 since that will deal with unreporting

**Comment by madmax330** [comment]
Still on hold

**Comment by trjExpensify** [comment]
As we chatted about [here](https://expensify.slack.com/archives/C036QM0SLJK/p1713366747773359?thread_ts=1713194515.544489&cid=C036QM0SLJK), taking it off hold.

**Comment by madmax330** [comment]
Picking this back up today, moving it back to daily

**Comment by madmax330** [comment]
This was deployed 4 days ago 😄 https://github.com/Expensify/Auth/pull/10642#issuecomment-2091505046

**Comment by trjExpensify** [comment]
Shall we close this @madmax330?

**Comment by madmax330** [comment]
This is done
